python/sglang/srt/managers/tokenizer_manager.py

```diff
@@ -712,21 +713,41 @@ async def init_weights_update_group(
         result = (await self.init_weights_update_group_communicator(obj))[0]
         return result.success, result.message
 
-    async def update_weights_from_distributed(
+    async def batch_update_weights_from_distributed(
         self,
-        obj: UpdateWeightsFromDistributedReqInput,
+        obj: BatchUpdateWeightsFromDistributedReqInput,
         request: Optional[fastapi.Request] = None,
     ) -> Tuple[bool, str]:
+        """Update multiple model parameters from distributed in a single request."""
         self.auto_create_handle_loop()
         assert (
             self.server_args.dp_size == 1
-        ), "dp_size must be for update weights from distributed"
+        ), "dp_size must be 1 for update weights from distributed"
 
-        # This means that weight sync
-        # cannot run while requests are in progress.
+        # Get a writer lock for the entire batch update operation
         async with self.model_update_lock.writer_lock:
-            result = (await self.update_weights_from_distributed_communicator(obj))[0]
-            return result.success, result.message
+            all_success = True
```

Comment:
nit: seems we can remove this variable, and use `len(error_messages) == 0` to know it succeeds